### PR TITLE
Auto-generates DB tables if they did not exists before

### PR DESCRIPTION
**Pull request by anthonymonori**
Scenario: When the user downloads the application and only has an empty database with no knowledge of the schema, the app fails and hard to debug the schema.

This commit solves this issues by auto-generating the tables based on an sql script and goes ahead as usual.

### CODE REVIEW

**Comment by anthonymonori** [comment]
Could any of you review the code and let us know in the comments if it is 👍or 👎?!

File: BusinessLogic/src/main/java/BusinessLogicLayer/EnronSearchEngine.java [hunk 1]
```diff
@@ -58,6 +69,51 @@ public static void main(String[] args) throws Exception {
         shutdownAndAwaitTermination(pool);
     }
 
+    private static boolean checkTables() {
+        String sqlCheck = "SHOW TABLES";
+        try (Connection connection = Database.getConnection()) {
+            PreparedStatement ps = connection.prepareStatement(sqlCheck);
+            ResultSet res = ps.executeQuery();
```
Comment (hass123uk) [inline BusinessLogic/src/main/java/BusinessLogicLayer/EnronSearchEngine.java]: Both Prepared Statment and Result Set should also be included within the parameters of the try with resources block.

**Comment by hass123uk** [comment]
The changes should be moved to a class within the DAL.

**Comment by anthonymonori** [comment]
Should be OK to merge.

**Pull request merged**